reference/src/layout/enums.md

```diff
@@ -295,49 +295,43 @@ apply, as described below.
 
 #### Discriminant elision on Option-like enums
 
-(Meta-note: The content in this section is not fully described by any RFC and is
-therefore "non-normative". Parts of it were specified in
-[rust-lang/rust#60300]).
-
-[rust-lang/rust#60300]: https://github.com/rust-lang/rust/pull/60300
+(Meta-note: The content in this section have been turned into stable guarantees
+[via this
+FCP](https://github.com/rust-lang/rust/pull/130628#issuecomment-2402761599).).
 
 **Definition.** An **option-like enum** is a 2-variant `enum` where:
 
 - the `enum` has no explicit `#[repr(...)]`, and
-- one variant has a single field, and
-- the other variant has no fields (the "unit variant").
+- one variant has a single field with a type that guarantees discriminant
+  elision (to be defined below), and
```

Comment:
Restricting the definition of an option-like enum based on the type of the single field is quite strange to me. Because I'd think that excludes `Option<T>` from the definition, because `T` could be `usize` or any type without guaranteed discriminant elision.

Reply:
@saethlin does this answer your question? I don't think I quite understood the question, TBH.

Reply:
Whoops, I didn't notice you'd responded.

The text here is just confusing to me, because what I'm imagining is a reader saying "Is this an option-like enum":
```rust
enum Thing<T> {
    Yes(T),
    No,
}
```
Because by the text above, `Thing<T>` is not "an option-like enum" because its single field does not guarantee discriminant elision. This would be though:
```rust
enum Thing {
    Yes(NonZeroUsize),
    No,
}
```

...but *directly below this* the reference already says that `Option<T>` is an example of an option-like enum.

:shrug: 

I think you're making an assumption that the reader will interpret words a bit differently when a generic parameter is included. If other can confirm that they have read this and don't find the wording confusing I'm happy to be overruled.

Reply:
For this reason, in my type-layout docs PR, I chose to separate the idea of "Is an enum for which discriminant elision may apply" (in my text, called a *discriminant elision elible*)  and "A field of this type is subject to discriminant elision" (called *elision candidate type*). When the two are combined is when the result is guaranteed.
I also note explicitly that the determiniation is post-mono. and explicitly call out both `Option<T>` and `Result<T,E>`.

Reply:
I am implicitly assuming that we are talking about a fully monomorphic type here. It doesn't make sense to ask about the layout of a generic type. I can make that more explicit (though this will come at the cost of readability). I don't see how splitting this concept into two separate concepts is helpful.